service/lib/agama/autoyast/files_reader.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2024] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+# :nodoc:
+module Agama
+  module AutoYaST
+    # Builds the Agama "files" section from an AutoYaST profile.
+    class FilesReader
+      # @param profile [ProfileHash] AutoYaST profile
+      def initialize(profile)
+        @profile = profile
+      end
+
+      # Returns a hash corresponding to Agama "files" section.
+      #
+      # If there is no files-related information, it returns an empty hash.
+      #
+      # @return [Hash] Agama "files" section
+      def read
+        return {} if profile["files"].nil?
+
+        # files section contains list of file nodes.
+        # supported file sub-elements:
+        # - file_location (a file source path) (-> source)
+        # - file_contents (a file content) (-> source)
+        # - file_path (this one seems to be the only mandatory,
+        #             a file destination or directory to be created if ends with '/')
+        #             (-> destination)
+        # - file_owner (-> owner)
+        # - file_permissions (permissions)
+        #
+        # unsupported for now:
+        # - file_script
+        files = profile.fetch_as_array("files")
+
+        files_json = files.reduce([]) do |res, f|
+          file = file_source(f)
+
+          file["destination"] = f["file_path"] if f["file_path"]
+          file["permissions"] = f["file_permissions"] if f["file_permissions"]
+          file["owner"] = f["file_owner"] if f["file_owner"]
```

Comment:
as mentioned we will need to translate dot into colon as agama will use "root:root" and autoyast according to docu I found uses "root.root" see https://documentation.suse.com/sles/15-SP5/html/SLES-all/cha-configuration-installation-options.html#id-1.7.5.2.38.11

Reply:
as discussed with imo files won't have owner field and instead there will be user and group field.

Reply:
see https://github.com/agama-project/agama/pull/2121/commits/939ed07e7392c97d0d03494c9617e4174c4a204d

Reply:
so instead of ```{ "owner": "user:group" }``` add two json members ```{ "user": "user", "group": "group" }```? Ok, will update accordingly